src/plotrecipes.jl

```diff
@@ -1,6 +1,98 @@
 
 @recipe function f{T<:TimeArray}(ta::T)
-    labels --> reshape(ta.colnames,1,length(ta.colnames))
-    seriestype --> :path
-    ta.timestamp, ta.values
+    st = get(d, :seriestype, :path)
+    if in(st, [:candlestick, :heikinashi])
+        Candlestick(ta)
+    #elseif st == :ohlc #ohlc (meaning sticks with steps on the sides) should be passed on to Plots internal ohlc plot engine
+    #    ta, ohlc = extract_ohlc(ta)
+    #    collect(zip(ohlc)) # But there are currently issues with that
+    else
+        labels --> reshape(ta.colnames,1,length(ta.colnames))
+        seriestype := st
+        ta.timestamp, ta.values
+    end
+end
+
+type Candlestick{D <: TimeType}
+    time::Vector{D}
+    open::AbstractVector
+    high::AbstractVector
+    low::AbstractVector
+    close::AbstractVector
+end
+
+Candlestick(ta::TimeArray) = Candlestick(extract_ohlc(ta)...)
+
```

Comment:
These two functions could be written tests for. It will be awhile before I find time to look at this again, but I don't feel strongly it has to be me who writes them.